modules/ppcp-api-client/src/Factory/ShippingFactory.php

```diff
@@ -96,30 +96,30 @@ public function from_wc_order( \WC_Order $order ): Shipping {
 	 * @throws RuntimeException When JSON object is malformed.
 	 */
 	public function from_paypal_response( \stdClass $data ): Shipping {
-		if ( ! isset( $data->name->full_name ) ) {
-			throw new RuntimeException( 'No name was given for shipping.' );
+		$name = $data->name->full_name ?? null;
+
+		$address = null;
+		if ( isset( $data->address ) ) {
+			$address = $this->address_factory->from_paypal_response( $data->address );
 		}
-		if ( ! isset( $data->address ) ) {
-			throw new RuntimeException( 'No address was given for shipping.' );
+
+		$contact_email = null;
+		if ( isset( $data->email_address ) ) {
+			$contact_email = $data->email_address;
 		}
```

Comment:
This can be simplified

```suggestion
		$contact_email = $data->email_address ?? null;
```